api/jobs_test.go

```diff
@@ -690,3 +698,481 @@ func TestPutNumTasksHandler(t *testing.T) {
+				Path:  models.JobStatePath,
+				Value: models.JobStateFailed,
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, failedStatePatches, currentJob, log.Data{})
+
+			Convey("Then updatedJob and bsonUpdates should contain updates from the patch", func() {
+				So(updatedJob.State, ShouldEqual, models.JobStateFailed)
+				So(bsonUpdates[models.JobStateBSONKey], ShouldEqual, models.JobStateFailed)
+
+				Convey("And ReindexFailed should be updated", func() {
+					So(updatedJob.ReindexFailed, ShouldNotEqual, currentJob.ReindexFailed)
+					So(bsonUpdates[models.JobReindexFailedBSONKey], ShouldNotBeEmpty)
+
+					Convey("And LastUpdated should be updated", func() {
+						So(updatedJob.LastUpdated, ShouldNotEqual, currentJob.LastUpdated)
+						So(bsonUpdates[models.JobLastUpdatedBSONKey], ShouldNotBeEmpty)
+
+						Convey("And no error should be returned", func() {
+							So(err, ShouldBeNil)
+						})
+					})
+				})
+			})
+		})
+	})
+
+	Convey("Given patches which changes state to completed", t, func() {
+		completedStatePatches := []dprequest.Patch{
+			{
+				Op:    dprequest.OpReplace.String(),
+				Path:  models.JobStatePath,
+				Value: models.JobStateCompleted,
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, completedStatePatches, currentJob, log.Data{})
+
+			Convey("Then updatedJob and bsonUpdates should contain updates from the patch", func() {
+				So(updatedJob.State, ShouldEqual, models.JobStateCompleted)
+				So(bsonUpdates[models.JobStateBSONKey], ShouldEqual, models.JobStateCompleted)
+
+				Convey("And ReindexCompleted should be updated", func() {
+					So(updatedJob.ReindexCompleted, ShouldNotEqual, currentJob.ReindexCompleted)
+					So(bsonUpdates[models.JobReindexCompletedBSONKey], ShouldNotBeEmpty)
+
+					Convey("And LastUpdated should be updated", func() {
+						So(updatedJob.LastUpdated, ShouldNotEqual, currentJob.LastUpdated)
+						So(bsonUpdates[models.JobLastUpdatedBSONKey], ShouldNotBeEmpty)
+
+						Convey("And no error should be returned", func() {
+							So(err, ShouldBeNil)
+						})
+					})
+				})
+			})
+		})
+	})
+}
+
+func TestPreparePatchUpdatesFail(t *testing.T) {
+	t.Parallel()
+	testCtx := context.Background()
+
+	currentJob, err := models.NewJob(validJobID1)
+	if err != nil {
+		t.Error("failed to get job to test preparePatchUpdates")
+	}
+
+	Convey("Given patches with unknown patch operation", t, func() {
+		unknownOpPatches := []dprequest.Patch{
+			{
+				Op:   dprequest.OpRemove.String(),
+				Path: models.JobTotalSearchDocumentsPath,
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, unknownOpPatches, currentJob, log.Data{})
+
+			Convey("And an error should be returned", func() {
```

Comment:
```suggestion
			Convey("Then an error should be returned", func() {
```

Reply:
Sorry about that. Thank you for spotting it